camelot/view/validator.py

```diff
@@ -71,3 +81,113 @@ def validate(self, input_, pos):
         except ParsingError:
             return (QtGui.QValidator.State.Intermediate, input_, pos)
         return (QtGui.QValidator.State.Acceptable, input_, pos)
+
+class ZipcodeValidator(QtGui.QValidator, AbstractValidator):
+
+    @dataclass    
+    class State(DataclassSerializable):
+
+        regex: str = None
+        regex_repl: str = None
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.state = self.State()
+
+    def set_state(self, state):
+        state = state or dict()
+        assert isinstance(state, dict)
+        self.state = self.State(**state)
```

Comment:
we kunnen best hier de regexpen al compileren ?

Reply:
Mja, voor de hoofdregex is dat inderdaad een optie, maar het replace pattern is in feite geen klassieke regex, dus dat op zichzelf compileren geeft fouten (de groepreferenties worden dan immers niet gevonden). De replace kan enkel worden geinterpreteerd in combinatie met de hoofdregex.

Reply:
Fixed in https://github.com/v-finance/camelot/pull/609/commits/3ba0f88388cdd427c0d19467bfea41b19adfbd1c.